### DISCUSSION TITLE
Is there a fast replacement of exatermux?

### DISCUSSION THREAD

**Discussion by erfty125** [termux/termux-app#3042] [Q&A] [answered]
I just want to compile with ndk fast. I heard about wine but it not supported. Also im using ndk r10e.

**Comment by agnostic-apollo** [comment, marked as answer]
What is exatermux? What do you need to compile? Also termux is using `25b` currently so won't provide support now for something as old as `r10e`.

Packages for termux can be compiled as per instructions at https://github.com/termux/termux-packages/wiki/Building-packages

If you still want to cross compile something small/random with `r10e`, you can try https://github.com/agnostic-apollo/Android-NDK-Cross-Compile-Build-Automator

**Reply by erfty125** [reply]
exatermux emulates the required architecture for ndk (arch 64). I didn’t have any problems with cross-compiling because everything was already set up in hlsdk portable (yes, I want to build hlsdk-protable) and it also requires ndk r10e because there may be bugs on the new ones. I actually need to emulate 64 bit so that ndk can compile everything (I have a 32 bit system, hardfp) . Also , Do I need to compile any packages?

**Reply by erfty125** [reply]
> What is exatermux? What do you need to compile? Also termux is using `25b` currently so won't provide support now for something as old as `r10e`.
 Does that means that i dont need to emulate 64 bit architecture for ndk 25b?

**Reply by agnostic-apollo** [reply]
>Exagear is a virtualization technology makes it possible to run Intel x86 applications on ARM-based devices... but was again resumed in 2020 under the Huawei brand and can already run x86_64 applications on ARM64 devices.

https://github.com/ZhymabekRoman/Exagear-For-Termux

https://news.ycombinator.com/item?id=25749490

It doesn't seem to emulate (aarch) 64 on 32bit hardware, so not sure what you are trying to do and not really related to termux anyways.

> Does that means that i dont need to emulate 64 bit architecture for ndk 25b?

Newer ndks require 64bit linux, windows or macos to run. Available for android 64bit as well.

https://developer.android.com/ndk/downloads 

https://github.com/Lzhiyong/termux-ndk

**Reply by erfty125** [reply]
My bad , i need to emulate i386 architecture*
